Author show

treble37 opened the pull request:
Implemented the first part of this feature (https://github.com/ChillinAstroTurtles/astro-reads/issues/13) completely.    Still need to show the sample covers.

treble37 committed:
Create Author model

- Write tests that check for author object properties

treble37 committed:
Add functionality for author show page

- Add authors_controller.rb
- Add has_many :through relationship so books can have co-authors
- Add show and books route for authors controller

treble37 committed:
Display an author's show page

- List an author's name, bio, etc.
- List an author's books, up to 5
- Updated seeds file to allow UI testing on development
- Update authors_controller and routes.rb to the books route with an :id
  param

- Add Faker gem to test group so CircleCI tests will pass

- Omit space at class body end in authors_controller

dstrunk commented on app/models/book_author.rb:
```diff
@@ -0,0 +1,4 @@
+class BookAuthor < ActiveRecord::Base
```
I don't think we need a `through` table here... unless you're thinking a book _should_ have many authors? For me, it makes more sense to have `book belongs_to :author`, and `author has_many :books`. Having a through table seems like premature normalization, and I don't see us needing to reference the BookAuthor relationship _directly_.

If we do think a join table is necessary, can we consider a different name? Something like "Credit" which is easier to hold in our heads.

treble37 replied:
Thanks @dstrunk, I probably should have run this by you guys first.  I was thinking about the case of co-authors: https://www.goodreads.com/book/show/67035.Relic?from_search=true&search_version=service  ( Douglas Preston (Goodreads Author), Lincoln Child) - what do you think?

dstrunk replied:
That's fine! At first glance it just looked like premature optimization.

What do you think about renaming the model to "Credit" to make more sense from a vocabulary standpoint?

treble37 replied:
Naming is hard...credit actually made me think of it in a financial sense before I thought about it further....but I'm fine with it if the group decides that's a name for it.  Maybe we should talk about it on the call tonight.

dstrunk commented on config/routes.rb:
```diff
@@ -1,3 +1,5 @@
 Rails.application.routes.draw do
   resources :books, only: [:index]
+  resources :authors, only: [:show]
+  get "/authors/:id/books", to: "authors#books"
```
This non-RESTful route might be uneccessary. For an MVP, and following the specs on Upcase, I think it's fine to just render up to 5 books on the Author `show` page (perhaps in a book partial, after initially implemented without a partial).

treble37 replied:
Thanks @dstrunk, good catch.  I'll refactor it out and make a commit showing the edit.

treble37 committed:
Rollback has_many :through relationship

- Delete book_author relationship since group decided to replace
  has_many :through relationship between books and authors with a
  has_and_belongs_to_many
- Update author_spec.rb and book_spec.rb tests

treble37 committed:
Update seeds file

- Update seeds.rb to deal with new has_and_belongs_to_many relationship
  between Book and Author models

treble37 committed:
Update Author page

- Author show page will now show books (limit 5)
- Remove non-restful route "books"
- Remove associated controller method with route "books" from
  authors_controller.rb

treble37 committed:
Add cover_image_url to Book

- Update seeds.rb for creating Book models
- Update Author show page to display cover_image_url of book
- Fix Hound CI violations

treble37 commented:
@dstrunk , @mrlee-io  - please take a look at item 2 below especially:

1) Added cover_image_url to Book
2)  Hound CI is complaining about preferring has_many :through - should we disregard for now and merge this in?

treble37 committed:
Update array syntax in seeds.rb

- Omit quotations marks and commas per
  http://www.zenspider.com/Languages/Ruby/QuickRef.html#arrays

treble37 commented:
@dstrunk, thanks I fixed the array, the thoughtbot guide says to avoid collection/member....with that, are you ok to merge then?

dstrunk commented:
:+1:
